src/lib/eslint/BaseEslintEngine.ts

```diff
@@ -78,6 +76,7 @@ export abstract class BaseEslintEngine implements RuleEngine {
 	private initializedBase: boolean;
 	protected outputProcessor: OutputProcessor;
 	private baseDependencies: StaticDependencies;
+	private config: Config;
```

Comment:
Moved some common code up from the strategies. Use the ENGINE enum instead of the string for identification.